web/src/components/apps/AppDetailPage.tsx

```diff
@@ -95,9 +92,14 @@ function AppDetailPage(props: Props) {
 
   const history = useHistory();
   const params = useParams<KotsParams>();
-  const [currentApp, setCurrentApp] = useState(
-    props.appsList?.find((w) => w.slug === params.slug)
-  );
+  const { selectedApp } = useSelectedApp();
+  const {
+    data: appsData,
+    refetch: refetchApps,
+    isLoading: appsIsLoading,
+  } = useApps();
+
+  const { apps: appsList } = appsData || {};
```

Comment:
for now the custom hooks just return exactly what `react-query` returns. that means that each component that uses the hook has to handle the response possibly being `null`